packages/snaps-execution-environments/src/common/BaseSnapExecutor.ts

```diff
@@ -396,9 +394,12 @@ export class BaseSnapExecutor {
 
     const request = async (args: RequestArguments) => {
       assertSnapOutboundRequest(args);
+      const sanitizedArgs = sanitizeJsonStructure(
+        args,
+      ) as unknown as RequestArguments;
```

Comment:
why is the casting to `unknown` needed before casting to `RequestArguments`?

Reply:
This is because `RequestArguments` is more strictly defined type and `Json` does not overlap with it.

```perl
TS2352: Conversion of type 'string | number | boolean | Json[] | { [prop: string]: Json; } | null' to type 'RequestArguments' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.   Property 'method' is missing in type '{ [prop: string]: Json; }' but required in type 'RequestArguments'.
```

Given that the `RequestArguments` is plain object its integrity should not be affected after validation and JSON stringify and parse processing.


```typescript
export interface RequestArguments {
    /** The RPC method to request. */
    method: string;
    /** The params of the RPC method, if any. */
    params?: unknown[] | Record<string, unknown>;
}
```

Let me know if you see anything wrong here. Thanks!

Reply:
This will be resolved after updating https://github.com/MetaMask/utils/pull/103 which is now using generic types.